pyteal/ast/subroutine.py

```diff
@@ -282,7 +282,17 @@ def invoke(
                     raise TealInputError(
                         f"supplied argument at index {i} should be an ABI type but got {arg}"
                     )
-                if arg.type_spec() != arg_type:
+
+                # If the expected arg is a TransactionType we can relax our check
+                # to be any subclass of TransactionTypeSpec
+                if arg_type.__class__ is abi.TransactionTypeSpec:
```

Comment:
nit: would prefer code style to be
```suggestion
                if type(arg_type) is abi.TransactionTypeSpec:
```